pkg/sanity/controller.go

```diff
@@ -197,32 +197,7 @@ var _ = DescribeSanity("Controller Service", func(sc *SanityContext) {
 			Expect(serverError.Code()).To(Equal(codes.Aborted))
 		})
 
-		It("should fail when the starting_token is greater than total number of vols", func() {
```

Comment:
Why was your driver failing this test? `<totalVols + 5>` isn't a valid starting token, or is it?

Would it be better to pass a random UUID?

Reply:
This test is incorrect because the CSI spec defines tokens as strings, and these strings may not be representations of integers (spec doesn't say anything about the token format, so these are plugin specific).

And even *if* they were integers, they may not indicate the numbers of created volumes. For example they could be integers representing a timestamp.

So the only thing we can tell about tokens is that a weird string will probably fail (we test this in the previous test), and that a token returned by the plugin must be acceptable and not fail when passes as starting point for the next pagination request, even if the volumes have changed.

Reply:
> This test is incorrect because the CSI spec defines tokens as strings, and these strings may not be representations of integers

But the spec also doesn't say that the token must not be the representation of integers. The wording of the test isn't good, but the check itself makes sense.

The test should be called "should fail when the starting_token is invalid", and then we need a good way of creating an invalid token.

Reply:
That is correct, the spec doesn't say that the token cannot be an integer.  In Ember-CSI our tokens are integers representing the volume creation time so we can use them to filter volumes and make sure we don't return duplicates and always return newer volumes.

The test you mention already exists, it's the one before this one, L185-L198.  Although it is possible that "invalid-token" could be a valid token for some plugin.

Reply:
> The test you mention already exists, it's the one before this one

Oh, right. In that case I agree, "should fail when the starting_token is greater than total number of vols" can be removed. It might be a relevant test case for the CSI hostpath driver, but that's no reason to have it here.

The risk is too high that the integer actually is treated as a valid token by some driver, which then fails the test. Is that what happened with your driver?

Reply:
Exactly, any integer number is a valid token for Ember-CSI.